R/tostring.R

```diff
@@ -154,6 +186,32 @@ setMethod("toString", "MatrixPrintForm", function(x,
 ##     x
 ## }
 
+splitter_new_line <- function(x, nchar_max = 80, txt_indent = 1) {
+  gen_tmp_list <- list(x)
+  while (TRUE) {
+    tmp_split <- unlist(strsplit(gen_tmp_list[[length(gen_tmp_list)]],
+      split = " ",
+      fixed = TRUE
+    ))
+    to_keep <- cumsum(sapply(tmp_split, function(y) nchar(y) + 1)) < nchar_max
+    if (isFALSE(to_keep[1])) {
+      err_msg <- paste(
+        "Indenting title or footnotes: words are too long or nchar_max is too small.",
+        "We suggest using shorter words, or increasing the relative parameters or",
+        "add manually the new line symbol."
+      )
+      stop(err_msg)
+    }
+    if (all(to_keep)) break
+    gen_tmp_list[[length(gen_tmp_list)]] <- tmp_split[to_keep]
+    gen_tmp_list[[length(gen_tmp_list) + 1]] <- tmp_split[!to_keep]
+  }
+  paste0(sapply(gen_tmp_list, function(y) {
+    paste0(y, collapse = " ")
+  }), collapse = paste0(c("\n", rep(" ", txt_indent)), collapse = ""))
```

Comment:
IF we are going to do this, we already have utilities to indent and pad strings, I believe, please use these (again, if we implement this functionality ourselves at all, which i hope not to)